controllers/control_plane_init_locker.go

```diff
@@ -77,7 +80,7 @@ func (l *controlPlaneInitLocker) Acquire(cluster *clusterv2.Cluster) bool {
 	}
 
 	log.Info("Attempting to create control plane configmap lock")
-	_, err = l.configMapClient.ConfigMaps(cluster.Namespace).Create(controlPlaneConfigMap)
+	err = l.client.Create(l.ctx, controlPlaneConfigMap)
 	if err != nil {
```

Comment:
Nit: prefer using `if err := <operation>; err != nil {` which scopes the variable `err` only to the following block